sky/shell/BUILD.gn

```diff
@@ -220,6 +220,35 @@ if (is_android) {
       ":sky_shell",
     ]
   }
+
+  # The shell_lib target is used by the 'flutter apk' tool from the flutter
+  # repository.
+  standalone_lib = "$target_gen_dir/standalone_lib.dex.jar"
+  android_standalone_library("standalone_lib") {
+    dex_path = standalone_lib
+    deps = [
+      ":java",
+    ]
+  }
+
+  dex("shell_lib") {
+    deps = [
+      ":standalone_lib",
+      ":shell_apk",
+    ]
+    sources = [
+      standalone_lib,
+      "$target_gen_dir/shell_apk/shell_apk.jar",
+    ]
+    output = "$target_gen_dir/shell_lib.dex.jar"
+  }
+
+  group("shell") {
+    deps = [
+      ":shell_apk",
+      ":shell_lib",
+    ]
+  }
```

Comment:
Good suggestion. I changed the android_apk command from our rules.gni, and now it builds a classes.dex.jar before linking it into classes.dex. The .dex.jar is used by both our build system and by 'flutter apk' when linking with other services. Seems much cleaner now.

jar vs dex.jar - jar contains the java bytecode (bunch of .class files), while dex.jar has the single dalvik binary. The main tradeoff that I can see is that dex.jars seem to be smaller. But there's probably others that I don't know about.